pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -289,6 +289,9 @@ def __add__(self, H):
         ops = self.ops.copy()
         coeffs = self.coeffs.copy()
 
+        if isinstance(H, qml.pulse.RydbergHamiltonian):
+            return H.__radd__(self)
```

Comment:
A `RydbergHamiltonian` is a `ParametrizedHamiltonian`, so in line 295, if we don't add this, the logic in the next if-clause will be executed and we will get a `ParametrizedHamiltonian` with all the correct coefficients, but none of the information about pulses or ordering of parameters. 

Basically this is a way of redirecting to the logic about adding a `RydbergHamiltonian` and a `ParametrizedHamiltonian`. This way all everything specific to adding `RydbergHamiltonian`s with other things lives in `rydberg_hamiltonian.py`.

An alternative to this would be to include a modified version of lines 353 to 372 of `rydberg_hamiltonian.py` here. Perhaps this would be clearer?

Reply:
I'm wrong, this does nothing and I'm deleting it 😅